src/lib/validate.js

```diff
@@ -67,14 +67,51 @@ module.exports.validatePassword = (password, username) => {
     // https://stackoverflow.com/a/54370584/2308190
     } else if (Array.from(password).length < 6) {
         return {valid: false, errMsgId: 'registration.validationPasswordLength'};
-    } else if (password === 'password') {
+    } else if (password.toLowerCase() === 'password') {
         return {valid: false, errMsgId: 'registration.validationPasswordNotEquals'};
     } else if (username && password === username) {
         return {valid: false, errMsgId: 'registration.validationPasswordNotUsername'};
     }
     return {valid: true};
 };
 
+module.exports.validatePasswordRemotely = password => (
+    new Promise(resolve => {
+        api({
+            method: 'POST',
+            uri: `/accounts/checkpassword`,
+            json: {
```

Comment:
Is this as secure as we can reasonably be with the password value? Will HTTPS basically protect this info? My knowledge is a little shaky here, but am I right in understanding that a password HTML element value will have a surface level of encryption applied when submitting via POST -- is that level of encryption better?

Reply:
Yes. HTTPS is already protecting the password info during the actual sending of the password when setting up the join/change password process.

Reply:
The reason to do a POST and/or a PUT is that GETs are technically not supposed to have a body's according to the RFCs and many libraries in the world will either not support sending a body or outright respond with an error if they see one.

Reply:
And putting the password in the querystring would be insecure becuase the actual URL is more visible than the POST payload?